File: a695(prime_factor)/C++/prime_factor.cpp
#include <iostream>
#include <cmath>

bool is_prime(int number) {
  if(number == 2 || number == 3) {
    return true;
  } else if(number % 6 != 1 && number % 6 != 5) {
      return false;
  } else {
    int ceil = sqrt(number);
    for(int i = 5; i <= ceil; i += 6) {
      if(number % i == 0 || number % (i+2) == 0) {
        return false;
      }
    }
  }
  return number != 1;
}

int main(int argc, char const *argv[]) {
  int number;
  while (std::cin >> number) {
    if (is_prime(number)) {
      std::cout << number << '\n';
      continue;
    }
    for (int n = 2; n < (int)sqrt(number) + 1; n++) {
      if (number % n == 0 && is_prime(n)) {
        std::cout << number / n << '\n';
        break;
      }
    }
  }
  return 0;
}

File: a695(prime_factor)/Python/prime_factor.py
def is_prime(number: int) -> bool:
    if number == 2 or number == 3:
        return True
    if number % 6 != 1 and number % 6 != 5:
        return False
    else:
        ceil = int(number ** 0.5)
        for n in range(5, ceil+1, 6):
            if number % n == 0 or number % (n+2) == 0:
                return False
    return number != 1


while True:
    try:
        number = int(input())
        if is_prime(number):
            print(number)
            continue
        for n in range(2, int(number ** 0.5) + 1):
            if number % n == 0 and is_prime(n):
                print(number // n)
                break
    except EOFError:
        break
